sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_message.py

```diff
@@ -0,0 +1,450 @@
+
+    :keyword data_body: The body consists of one or more data sections and each section contains opaque binary data.
+    :paramtype data_body: Optional[Union[str, bytes, List[Union[str, bytes]]]]
+    :keyword sequence_body: The body consists of one or more sequence sections and
+     each section contains an arbitrary number of structured data elements.
+    :paramtype sequence_body: Optional[List[Any]]
+    :keyword value_body: The body consists of one amqp-value section and the section contains a single AMQP value.
+    :paramtype value_body: Any
+    :keyword header: The amqp message header.
+    :paramtype header: Optional[~azure.servicebus.amqp.AMQPMessageHeader]
+    :keyword footer: The amqp message footer.
+    :paramtype footer: Optional[dict]
+    :keyword properties: Properties to add to the amqp message.
+    :paramtype properties: Optional[~azure.servicebus.amqp.AMQPMessageProperties]
+    :keyword application_properties: Service specific application properties.
+    :paramtype application_properties: Optional[dict]
+    :keyword annotations: Service specific message annotations.
+    :paramtype annotations: Optional[dict]
+    :keyword delivery_annotations: Service specific delivery annotations.
+    :paramtype delivery_annotations: Optional[dict]
+    """
+
+    def __init__(self, **kwargs):
+        # type: (Any) -> None
+        self._message = kwargs.pop("message", None)
+
+        if self._message:
+            # internal usage only for service bus received message
+            return
+
+        self._data_body = kwargs.pop("data_body", None)
+        self._sequence_body = kwargs.pop("sequence_body", None)
+        self._value_body = kwargs.pop("value_body", None)
+
+        validation = [
+            body
+            for body in (self._value_body, self._data_body, self._sequence_body)
+            if body is not None
+        ]
+        if len(validation) != 1:
+            raise ValueError(
+                "There should be one and only one of either data_body, sequence_body "
+                "or value_body being set as the body of the AMQPAnnotatedMessage."
+            )
+
+        header = kwargs.get("header")
+        footer = kwargs.get("footer")
+        properties = kwargs.get("properties")
+        application_properties = kwargs.get("application_properties")
+        annotations = kwargs.get("annotations")
+        delivery_annotations = kwargs.get("delivery_annotations")
+
+        body_type = (
+            uamqp.MessageBodyType.Data
+            if self._data_body
+            else (
+                uamqp.MessageBodyType.Sequence
+                if self._sequence_body
+                else uamqp.MessageBodyType.Value
+            )
+        )
+        body = self._data_body or self._sequence_body or self._value_body
+
+        message_header = None
+        message_properties = None
+        if header:
+            message_header = uamqp.message.MessageHeader()
```

Comment:
you probably know this already, but I think this will need to be replaced/wrapped by `AMQPMessageHeader()`?